nixos/tests/systemd-analyze.nix

```diff
@@ -0,0 +1,41 @@
+import ./make-test-python.nix ({ pkgs, latestKernel ? false, ... }:
+
+{
+  name = "systemd-analyze";
+  meta = with pkgs.stdenv.lib.maintainers; {
+    maintainers = [ raskin ];
+  };
+
+  machine =
+    { pkgs, lib, ... }:
+    { boot.kernelPackages = lib.mkIf latestKernel pkgs.linuxPackages_latest;
+      sound.enable = true; # needed for the factl test, /dev/snd/* exists without them but udev doesn't care then
+    };
+
+  testScript = ''
+    machine.wait_for_unit("multi-user.target")
+
+    with subtest("Prepare output dir"):
+        machine.succeed("mkdir systemd-analyze")
```

Comment:
I think we can put the equivalent of doing a `mkdir -p $(dirname $dst)` into the`copy_from_vm` function, so we don't have to repeat ourselves.

Also see `nixos/tests/blivet.nix`, which would benefit from that too (if it wouldn't have been broken, see https://github.com/NixOS/nixpkgs/issues/33496).

Reply:
This would be a change in #74077 of course, I can just better explain it with the example here.

Reply:
This is in-VM directory for `systemd-analyze` output that will later become the original source location for `copy_from_vm`